server/src/main/java/org/elasticsearch/search/builder/SearchSourceBuilder.java

```diff
@@ -292,7 +292,7 @@ public void writeTo(StreamOutput out) throws IOException {
         out.writeOptionalNamedWriteable(postQueryBuilder);
         if (out.getTransportVersion().onOrAfter(TransportVersion.V_8_500_013)) {
             out.writeList(subSearchSourceBuilders);
-        } else if (out.getTransportVersion().before(TransportVersion.V_8_8_0) && subSearchSourceBuilders.size() >= 2) {
+        } else if (out.getTransportVersion().before(TransportVersion.V_8_4_0) && subSearchSourceBuilders.size() >= 2) {
```

Comment:
this only works if one of the subsearches is kNN and one is a query. Two queries breaks here right?

Reply:
That is correct. However, I don't like the fragility of doing an `instanceof` check to ensure that one of these is a kNN query. Versions prior to kNN should break anyway when de-serializing the knn docs and score query. I view this as a courtesy to the user more than an actual exception as incorrect input here breaks the implicit contract of not using new features during upgrades that are unsupported.